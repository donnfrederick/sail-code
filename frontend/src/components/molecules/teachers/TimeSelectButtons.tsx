import FlatButton from 'components/atoms/teachers/FlatButton'
import * as ConversationModels from 'models/conversation'
import * as React from 'react'
import styled from 'styled-components'

interface Props {
  type: ConversationModels.TimeSelectType
  onClick(type: ConversationModels.TimeSelectType): void
}

export default (props: Props) => {
  const { type, onClick } = props
  return (
    <Container>
      <FlatButton
        text={`時刻で選ぶ`}
        isSelected={type === 'from'}
        width={344}
        height={68}
        fontSize={40}
        onClick={() => onClick('from')}
      />
      <FlatButton
        text={`時間帯で選ぶ`}
        isSelected={type === 'zone'}
        width={344}
        height={68}
        fontSize={40}
        onClick={() => onClick('zone')}
      />
    </Container>
  )
}

const Container = styled.div`
  display: flex;
  justify-content: space-between;
  width: 720px;
  margin: 0 auto 40px;
`
