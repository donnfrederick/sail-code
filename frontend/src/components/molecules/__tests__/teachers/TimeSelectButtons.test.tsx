import TimeSelectButtons from 'components/molecules/teachers/TimeSelectButtons'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

test('render <TimeSelectButtons />', () => {
  const tree = renderer
    .create(<TimeSelectButtons type={'from'} onClick={() => null} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
