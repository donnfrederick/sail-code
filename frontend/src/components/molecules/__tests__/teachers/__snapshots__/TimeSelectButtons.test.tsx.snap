// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <TimeSelectButtons /> 1`] = `
<div
  className="sc-htpNat fnrSTu"
>
  <button
    className="sc-bdVaJa bDSVsF"
    color="#138efd"
    data-disabled={false}
    data-selected={true}
    fontSize={40}
    height={68}
    onClick={[Function]}
    width={344}
  >
    時刻で選ぶ
  </button>
  <button
    className="sc-bdVaJa bDSVsF"
    color="#138efd"
    data-disabled={false}
    data-selected={false}
    fontSize={40}
    height={68}
    onClick={[Function]}
    width={344}
  >
    時間帯で選ぶ
  </button>
</div>
`;
